### PR TITLE
Add new section for Developer tools and resources

### PR DESCRIPTION
**Pull request by TucksonDev**
- Gathered all devs resources inside that new section: Chains, Useful addresses, Node providers, SDK and the tutorials link.
- Added some slight modifications to the pages that already existed, although all relevant information has been kept
- Added sections for development frameworks, web3 libraries, monitoring tools / block explorers, and debugging tools.

Preview is [here](https://nitro-docs-git-add-developer-tools-and-resources-offchain-labs.vercel.app/for-devs/dev-tools-and-resources/overview).

### CODE REVIEW

**Comment by symbolpunk** [comment]
(Feel free to nudge me for an editorial review after SME approval if you'd like 🤙)

File: arbitrum-docs/for-devs/dev-tools-and-resources/debugging-tools.mdx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+### Tenderly
+
+[Tenderly](https://tenderly.co/) is an all-in-one Web3 development platform that empowers developers to build, test, monitor, and operate smart contracts from inception to mass adoption. Tenderly's debugging options focus on providing developers with efficient and user-friendly tools to identify and fix smart contract bugs and production issues. The Debugger enables developers to inspect smart contracts by analyzing precise lines of code in a human-readable format. With Tenderly's Simulator, developers can play out specific historical transactions and current transaction outcomes before sending them on-chain, allowing them to change relevant parameters and source code to test and debug contracts. The platform streamlines the debugging process, saving time and resources while improving smart contract reliability.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/dev-tools-and-resources/debugging-tools.mdx:17]: Should we talk about Tenderly being buggy when it comes to custom Geth errors and link to this maybe?
https://www.notion.so/arbitrum/I-tried-to-create-a-retryable-ticket-but-the-transaction-reverted-on-L1-How-can-I-debug-the-issue-606f4ef84f884f4cb940ffc483e7b2ec#c5ded8ccec6543309fdf4086c8ef4214

Comment (TucksonDev) [reply arbitrum-docs/for-devs/dev-tools-and-resources/debugging-tools.mdx:17]: These are meant to be general descriptions of those tools, but that gives me an idea. We can create a "How to debug Arbitrum transactions" page, where we describe the process and mention what you just said. Then, from Tenderly's description, we can link to that How to.
The How to is likely to get more traffic from devs anyway, and that FAQ will be more useful there, imo. What do you think?

Comment (TucksonDev) [reply arbitrum-docs/for-devs/dev-tools-and-resources/debugging-tools.mdx:17]: Added a note on Tenderly's limitation. Thanks!

**Pull request merged**